test-suite/src/test/require-jdk8/org/scalajs/testsuite/javalib/IntegerTestOnJDK8.scala

```diff
@@ -0,0 +1,221 @@
+/*                     __                                               *\
+**     ________ ___   / /  ___      __ ____  Scala.js Test Suite        **
+**    / __/ __// _ | / /  / _ | __ / // __/  (c) 2013, LAMP/EPFL        **
+**  __\ \/ /__/ __ |/ /__/ __ |/_// /_\ \    http://scala-js.org/       **
+** /____/\___/_/ |_/____/_/ | |__/ /____/                               **
+**                          |/____/                                     **
+\*                                                                      */
+package org.scalajs.testsuite.javalib
+
+import org.scalajs.jasminetest.JasmineTest
+
+object IntegerTestOnJDK8 extends JasmineTest {
+
+  // Explicitly define these as `var`'s to avoid any compile-time constant folding
+  var MaxValue: Int = Int.MaxValue
+  var MinValue: Int = Int.MinValue
+
+  describe("java.lang.Integer on JDK8 or higher") {
+    it("Should be represented with logical 4 bytes") {
+      expect(Integer.BYTES).toEqual(4)
```

Comment:
This doesn't test anything, in fact. scalac will constant fold this value with that found on the JDK classpath anyway.